File: src/main/java/net/atos/zac/flowable/util/TaskUtil.java
/*
 * SPDX-FileCopyrightText: 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.flowable.util;

import static nl.info.zac.app.task.model.TaakStatus.AFGEROND;
import static nl.info.zac.app.task.model.TaakStatus.NIET_TOEGEKEND;
import static nl.info.zac.app.task.model.TaakStatus.TOEGEKEND;

import org.flowable.common.engine.api.scope.ScopeTypes;
import org.flowable.task.api.Task;
import org.flowable.task.api.TaskInfo;

import nl.info.zac.app.task.model.TaakStatus;

public class TaskUtil {
    public static boolean isOpen(final TaskInfo taskInfo) {
        return getTaakStatus(taskInfo) != AFGEROND;
    }

    public static boolean isCmmnTask(final TaskInfo taskInfo) {
        return ScopeTypes.CMMN.equals(taskInfo.getScopeType());
    }

    public static TaakStatus getTaakStatus(final TaskInfo taskInfo) {
        return taskInfo instanceof Task ? (taskInfo.getAssignee() == null ? NIET_TOEGEKEND : TOEGEKEND) : AFGEROND;
    }
}

File: src/main/kotlin/nl/info/zac/flowable/util/TaskUtil.kt
/*
 * SPDX-FileCopyrightText: 2024 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.flowable.util

import nl.info.zac.app.task.model.TaakStatus.AFGEROND
import nl.info.zac.app.task.model.TaakStatus.NIET_TOEGEKEND
import nl.info.zac.app.task.model.TaakStatus.TOEGEKEND
import org.flowable.common.engine.api.scope.ScopeTypes
import org.flowable.task.api.Task
import org.flowable.task.api.TaskInfo

fun TaskInfo.isOpen() = taakStatus() != AFGEROND

fun TaskInfo.isCmmnTask() = ScopeTypes.CMMN == scopeType

fun TaskInfo.taakStatus() = if (this is Task) (if (assignee == null) NIET_TOEGEKEND else TOEGEKEND) else AFGEROND
